include/fiction/algorithms/simulation/sidb/quickexact.hpp

```diff
@@ -0,0 +1,477 @@
+                    simulation_result.additional_simulation_parameters.emplace_back("base_number",
+                                                                                    static_cast<uint64_t>(3));
+                    this->three_state_simulation(charge_lyt_new, simulation_result);
+                }
+            }
+
+            // In the case with only one SiDB in the layout (due to external potentials or defects, this single SiDB can
+            // be neutrally or even positively charged.)
+            else if (number_of_SiDBs == 1)
+            {
+                if (three_state_simulation_required)
+                {
+                    charge_lyt.set_base_number(3);
+                }
+                else
+                {
+                    charge_lyt.set_base_number(2);
+                }
+
+                while (charge_lyt.get_charge_index().first < charge_lyt.get_max_charge_index())
+                {
+
+                    if (charge_lyt.is_physically_valid())
+                    {
+                        charge_distribution_surface<Lyt> charge_lyt_copy{charge_lyt};
+                        simulation_result.charge_distributions.push_back(charge_lyt_copy);
+                    }
+                    charge_lyt.increase_charge_index_by_one(
+                        false);  // "false" allows that the charge state of the dependent cell is automatically changed
+                                 // based on the new charge distribution.
+                }
+                if (charge_lyt.is_physically_valid())
+                {
+                    charge_distribution_surface<Lyt> charge_lyt_copy{charge_lyt};
+                    simulation_result.charge_distributions.push_back(charge_lyt_copy);
+                }
+            }
+            // If the layout consists of only detected negatively charged SiDBs, this scope is
+            // executed.
+            else if (all_sidbs_in_lyt_without_negative_detected_ones.empty() && number_of_SiDBs > 1)
+            {
+                charge_distribution_surface<Lyt> charge_lyt_copy{charge_lyt};
+                for (const auto& cell : detected_negative_sidbs)
+                {
+                    charge_lyt.adding_sidb_to_layout(cell, -1);
+                }
+                simulation_result.charge_distributions.push_back(charge_lyt_copy);
+            }
+        }
+        simulation_result.simulation_runtime = time_counter;
+        result                               = simulation_result;
+    }
+
+    sidb_simulation_result<Lyt> get_simulation_results() const
+    {
+        return result;
+    }
+
+  private:
+    /**
+     * This function conducts 2-state physical simulation (negative, neutral).
+     *
+     * @param charge_lyt_new Initialized charge layout.
+     * @param sim_result sidb_simulation_result to collect gained results.
+     */
+    void two_state_simulation(charge_distribution_surface<Lyt>& charge_lyt_new, sidb_simulation_result<Lyt>& sim_result)
```

Comment:
Please rename `charge_lyt_new`.

Also, `sidb_simulation_result` can be an attribute of the class.